src/vulnerabilities/handler.js

```diff
@@ -0,0 +1,238 @@
+    return { status: 'complete' };
+  }
+
+  if (!isNonEmptyArray(vulnerabilityReport.vulnerableComponents)) {
+    // No vulnerabilities found
+    // Fetch opportunity
+    const { Opportunity } = dataAccess;
+    let opportunity;
+    try {
+      const opportunities = await Opportunity.allBySiteIdAndStatus(
+        site.getId(),
+        Oppty.STATUSES.NEW,
+      );
+      opportunity = opportunities.find((o) => o.getType() === AUDIT_TYPE);
+    } catch (e) {
+      log.error(`Fetching opportunities for siteId ${site.getId()} failed with error: ${e.message}`);
+      throw new Error(`Failed to fetch opportunities for siteId ${site.getId()}: ${e.message}`);
+    }
+
+    if (opportunity) {
+      // No vulnerabilities found, update opportunity status to RESOLVED
+      log.info(`[${AUDIT_TYPE}] [Site: ${site.getId()}] no vulnerabilities found, but found opportunity, updating status to RESOLVED`);
+      await opportunity.setStatus(Oppty.STATUSES.RESOLVED);
+
+      // We also need to update all suggestions inside this opportunity
+      // Get all suggestions for this opportunity
+      const suggestions = await opportunity.getSuggestions();
+
+      // If there are suggestions, update their status to outdated
+      if (isNonEmptyArray(suggestions)) {
+        await Suggestion.bulkUpdateStatus(suggestions, SuggestionDataAccess.STATUSES.FIXED);
+      }
+      opportunity.setUpdatedBy('system');
+      await opportunity.save();
+    }
+
+    return { status: 'complete' };
+  }
+
+  // Update opportunity
+  const opportunity = await convertToOpportunity(
+    auditUrl,
+    { siteId: auditData.siteId, id: auditData.auditId },
+    context,
+    createOpportunityData,
+    AUDIT_TYPE,
+    createOpportunityProps(auditData.auditResult.vulnerabilityReport),
+  );
+
+  if (!configuration.isHandlerEnabledForSite('security-vulnerabilities-auto-suggest', site)) {
+    log.info(`[${AUDIT_TYPE}] [Site: ${site.getId()}] security-vulnerabilities-auto-suggest not configured, skipping suggestion creation`);
+    return { status: 'complete' };
+  }
+
+  // As a buildKey we hash all the component details and add name and version for readability
+  const buildKey = (data) => {
+    const s = JSON.stringify(data);
+    let hash = 0;
+    for (let i = 0; i < s.length; i += 1) {
+      const code = s.charCodeAt(i);
+      // eslint-disable-next-line no-bitwise
+      hash = ((hash << 5) - hash) + code;
+      // eslint-disable-next-line no-bitwise
+      hash &= hash; // Convert to 32bit integer
+    }
+    return `${data.name}@${data.version}#${hash}`;
+  };
+
+  // Populate suggestions
+  await syncSuggestions({
+    opportunity,
+    newData: vulnerabilityReport.vulnerableComponents,
+    context,
+    buildKey,
+    mapNewSuggestion: (entry) => mapVulnerabilityToSuggestion(opportunity, entry),
+    log,
+  });
+
+  return { status: 'complete' };
+};
+
+export default new AuditBuilder()
+  .withUrlResolver(wwwUrlResolver)
```

Comment:
do we really need the URL resolver?
This one fetches RUM-related information but should not be needed for the VULN audit?